src/test/java/uk/gov/homeoffice/digital/sas/timecard/listeners/TimeEntryKafkaEntityListenerTest.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.homeoffice.digital.sas.timecard.listeners;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+@ExtendWith(MockitoExtension.class)
+class TimeEntryKafkaEntityListenerTest {
+
+  private final UUID ownerId = UUID.randomUUID();
+  private TimeEntry timeEntry;
+
+  @Mock
+  private KafkaProducerService<TimeEntry> kafkaProducerService;
+  private final TimeEntryKafkaEntityListener kafkaEntityListener =
+      new TimeEntryKafkaEntityListener();
+
+  @BeforeEach
+  void setup() {
+    timeEntry = new TimeEntry();
+    timeEntry.setOwnerId(ownerId);
+
+    kafkaEntityListener.createProducerService(kafkaProducerService);
+  }
+
+  @Test
+  void resolveMessageKey_timeEntryEntity_useOwnerIdAsMessageKey() {
```

Comment:
just on the wording of the test
In the validator, KafkaProducerServiceTest, jpaRest tests the last part is usually worded in past tense. As in what **should** have happened.
So I think this would become something like: `ownerIdReturnedAsMessageKey`
I think either way is correct, but better to maintain consistency.

Reply:
Updated